HRManagement/ManagementLogic/Management.cs

```diff
@@ -90,56 +90,149 @@ public List<Project> FindPoject(string keyword)
         //Lưu ý nên kiểm tra đối tượng lại một nữa xem đã hợp lệ chưa và add. ví dụ 
         public void Add(Employee e)
         {
-
+            if (!employeesList.Contains(e))
+            {
+                employeesList.Add(e);
+                SaveData();
+            }
         }
         public void Add(Department d)
         {
-
+            if (!departmentList.Contains(d))
+            {
+                departmentList.Add(d);
+                SaveData();
+            }
         }
         //Khi thêm một prọect mới thì cunngx phải thêm, leader và employees có thể bỏ trống
         //ếu có leaader hoặc một list emplyees thì phải đảm bảo Project đó đã được thêm cho các nhân viên đó .
         //Ví dụ bên duói 
         public void Add(Project d, Employee leader = null, List<Employee> employees = default)
         {
-            //giả sử đã tạo project thành comg
-            leader.AddProject(d);
-            foreach (Employee e in employees)
+            if (!projectList.Contains(d))
             {
-                e.AddProject(d);
+                projectList.Add(d);
+
+                if (leader != null)
+                {
+                    leader.AddProject(d); 
+                }
+
+                if (employees.Count > 0)
+                {
+                    foreach (Employee e in employees)
+                    {
+                        e.AddProject(d); 
+                    }
+                }
+
+                SaveData(); 
             }
         }
         public void Remove(Employee e)
         {
-
+            if (employeesList.Contains(e))
+            {
+                foreach (Project p in projectList)
+                {
+                    p.RemoveEmployee(e);
+                }
+                employeesList.Remove(e);
+                SaveData();
+            }
         }
         public void Remove(Department d)
         {
-
+            if (departmentList.Contains(d))
+            {
+                
+                foreach (Employee e in employeesList)

```

Comment:
sao xóa department mà duyệt list emplyees

Reply:
duyệt list employees để xóa luôn phòng ban liên quan đến nhân viên đó. Ví dụ như là m thuộc phòng IT, t xóa phòng IT thì đương nhiên m sẽ ko còn phòng IT nữa, nên phòng ban của m sẽ null